### PR TITLE
Refactor ipywidgets to its own module and enhancements

### PR DESCRIPTION
**Pull request by ChristianZaccaria**
# Issue link

Jira: https://issues.redhat.com/browse/RHOAIENG-13330 

# What changes have been made

- Refactor widgets to its own module
- Converted UI table widgets to its own class.
- Enhance widgets unit tests and increased coverage = 100%

# Verification steps

- e2e tests and UI notebooks tests should continue to pass.

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Comment by ChristianZaccaria** [comment]
@KPostOffice In the latest commit here, I've converted each individual widget from the table to its own class. Let me know hwhat you think, and if it meets with expectations, or if you had something else in mind wrt https://github.com/project-codeflare/codeflare-sdk/pull/681#discussion_r1773640000 Thank you!

**Review by KPostOffice — COMMENTED** [review]
Nice! I like it. Less args is nice. There's a failing unit test, but otherwise lgtm I think

File: tests/unit_test.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2022 IBM, Red Hat
+# Copyright 2024 IBM, Red Hat
```
Comment (Bobbins228) [inline tests/unit_test.py:1]: Not sure about this update. Doesn't this reflect creation date? (Shouldn't matter really we are getting rid of this file anyway)

**Review by Bobbins228 — APPROVED** [review]
Left a comment on changing the year of the copyright we can discuss some other time. 
Overall this looks good and I appreciate the comment changes :) 

/lgtm
/approve

File: tests/unit_test.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 import os
 import re
 import uuid
+from io import StringIO
```
Comment (Ygnas) [inline tests/unit_test.py:22]:
```suggestion

```
This file appears to have several unused imports. Would we want to remove those?

File: tests/unit_test.py (hunk 1 above)
Comment (ChristianZaccaria) [reply tests/unit_test.py]: Ohh so that's how it works, I reverted that now thanks!

File: tests/unit_test.py (hunk 2 above)
Comment (ChristianZaccaria) [reply tests/unit_test.py]: Good catch thanks! - I went ahead and removed all unused imports and variables from that file, and re-organised them.

**Pull request merged**